src/components/templates/PartyMap/components/RoomModal/RoomModal.tsx

```diff
@@ -20,15 +21,40 @@ import { RoomModalOngoingEvent, ScheduleItem } from "..";
 
 import "./RoomModal.scss";
 
-interface RoomModalProps {
+export type RoomModalProps = {
   onHide: () => void;
+  show: boolean;
+  currentVenue?: AnyVenue;
+  room?: Room;
+};
+
+export type RoomModalContentProps = {
   room: Room;
-}
+  currentVenueName: string;
+};
+
+export const RoomModal: React.FC<RoomModalProps> = ({
+  onHide,
+  room,
+  show,
+  currentVenue,
+}) => {
+  if (!room || !currentVenue) return null;
+
+  return (
+    <Modal show={show} onHide={onHide}>
+      <RoomModalContent room={room} currentVenueName={currentVenue.name} />
+    </Modal>
+  );
+};
 
-export const RoomModal: React.FC<RoomModalProps> = ({ onHide, room }) => {
+export const RoomModalContent: React.FC<RoomModalContentProps> = ({
+  room,
+  currentVenueName,
+}) => {
   const venueEvents = useSelector(venueEventsSelector) ?? [];
 
-  const { enterRoom, recentRoomUsers } = useRoom(room);
+  const { enterRoom, recentRoomUsers } = useRoom(room, currentVenueName);
 
   useMousetrap({
```

Comment:
I would still prefer having it here instead of <Map> `useKeyboardControls`.

1) We want to only have a listener when we need it.
2) It will require hacks to have `enterRoom` passed there since to have `enterRoom` we need to be sure there is selected room. (One way to do that would be to have noop fn returned from useRoom, but I don't want to go that way as it seems a bit hacky)

Reply:
From memory, the purpose of it within the `useKeyboardControls` was so that when you are on the map, and you come across a room, you can press enter to 'bring it up' (or maybe to enter it, or similar.. I'm not 100%, as this was keeping the existing legacy logic, even though I refactored the 'how' of it)

My desire for this not to be here is to avoid us 'spreading' keyboard control logic across the app.

A question to ask here is whether we specifically need this keyboard control at all, be it in `useKeyboardControls`, or in `RoomModal`. It somewhat feels like it doesn't do a lot for us. If our desire is that we can enter the room from the room modal when this dialog is open, it may just also be possible by using existing buttons/html features. eg. 'submitting a form', or focussing the 'enter' button so that if you press enter, that's the first thing that will be 'actioned'.

If we decide that we can't use basic html features, and that we need to keep it in `RoomModal`, then as a 'final resort', we will need to change it so that its not using a global binding, but instead is 'scoped' to only register within the containing `div`. We can get that ref with `useRef`, but I believe there is also a bug in `useMousetrap` currently (detailed either in `@debt` comments, or maybe in a backlog item; I can't remember specifically) where I don't believe it will work properly with a 'non-global' binding. So there will be a bugfix required for that as well.

Reply:
Let's keep the modal-keyboard thing convo here.

Reply:
https://github.com/sparkletown/internal-sparkle-issues/issues/303